test/kvtree_read_scatter_single_test.c

```diff
@@ -0,0 +1,127 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <kvtree.h>
+#include <kvtree_util.h>
+#include <mpi.h>
+#include <limits.h>
+
+/*
+ * Test reading back PREFIX* files into a single kvtree.  This verifies you
+ * can read back a complete PREFIX* set of files.  Also verify that if one
+ * of the PREFIX.*.* files is missing, that the function correctly fails.
+ */
+
+/* Create our PREFIX* files. */
+static int create_files(char *prefix_dir, int ranks)
+{
+  int rc;
+  kvtree *tree = kvtree_new();
+  char name[10];
+  char path[PATH_MAX];
+  int i;
+
+  /* Create a kvtree and stick some data in it */
+  for (i = 0; i < ranks; i++) {
+    sprintf(name, "%d", i);
+    kvtree_util_set_str(tree, name, "test data");
+  }
+
+  snprintf(path, sizeof(path), "%s/PREFIX", prefix_dir);
+  path[sizeof(path) - 1] = '\0';
+
+  rc = kvtree_write_to_gather(path, tree, ranks);
```

Comment:
To be certain that we handle that trailing ".0" case, let's add the following in create files so that it writes two sets of kvtree files:
```suggestion
  snprintf(path, sizeof(path), "%s/PREFIX", prefix_dir);
  path[sizeof(path) - 1] = '\0';
  rc = kvtree_write_to_gather(path, tree, ranks);

  snprintf(path, sizeof(path), "%s/PREFIX.0", prefix_dir);
  path[sizeof(path) - 1] = '\0';
  rc = kvtree_write_to_gather(path, tree, ranks);
```